magento/magento2
# Fix for issue 35789

### Description (*)
The problem was that the url_key was built on the basis of the product name and the product validation failed. It was decided to use the SKU to generate the url_key if there is no product name. If there are not both (issue-35810), then we generate a random url_key. Errors related to this has been fixed (grid and method $product->getName() expects a string, not null). 

### Fixed Issues (if relevant)
1. Fixes magento/magento2#35789

### Manual testing scenarios (*)
1. Login as admin.
2. Go to Store->attributes->product search code "name" and set Values Required to "NO".
3. Clear cacge
4. Go to catalog >> products >>Add Product.
5. Fill all mandatory details. Do not provide name.
6. Save the product - User is able to save the product without name
7. Create the product with Product name and save - Edit product - Remove product name and save - User is able to save the product without name
8. Check whether the product is visible in the grid
9. Check storefront

### Contribution checklist (*)
